pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,78 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+type AccessMode string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
+)
+
+// SubnetSpec defines the desired state of Subnet.
+type SubnetSpec struct {
+	// Size of Subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPv4SubnetSize int `json:"ipv4SubnetSize,omitempty"`
+	// Access mode of Subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	// +kubebuilder:validation:Enum=private;public;isolated
+	AccessMode AccessMode `json:"accessMode,omitempty"`
+	// Subnet CIDRS.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=2
+	IPAddresses []string `json:"ipAddresses,omitempty"`
+}
+
+// SubnetCondition defines condition of Subnet.
+type SubnetCondition struct {
+	// Type defines condition type.
+	Type SubnetStatusCondition `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about condition.
+	Message string `json:"message,omitempty"`
+}
+
+// SubnetStatus defines the observed state of Subnet.
+type SubnetStatus struct {
```

Comment:
@lxiaopei @heypnus : could you make sure Status structs use a common name (I now prefer "NSXResourcePath") for the resource path in NSX-T API.

And could you define a common Condition struct to be shared by all CRDs? I saw all CRD Condition definitions are the same. Let me know if I am wrong, and you believe we may have different condition definitions for different CRDs.

Reply:
Yes, We define a common Condition struct in this patch https://github.com/vmware-tanzu/nsx-operator/pull/108, after this patch be merged, all CRDs can use the same Condition. @jianjuns   Thanks for the reminder.

For the path under Status, all the CRDs will use NSXResourcePath.

Reply:
Thanks for clarifying, updated to use `NSXResourcePath`.